src/clr-angular/data/tree-view/models/tree-node.model.ts

```diff
@@ -55,6 +83,23 @@ export abstract class TreeNodeModel<T> {
     this.setSelected(newState, true, propagate);
   }
 
+  // To bypass the default event handlers and programmatically set the checked' & 'indeterminate' DOM properties of input checkbox
+  checkboxToggleHandler(propagate: boolean, event: Event) {
+    this.toggleSelection(propagate);
+    // When a user triggers the click event on checkbox inputs, checkbox has toggled its state from checked to unchecked or vice versa.
+    // And that does not get over written by the conditional operations being bound to 'checked' & 'indeterminate' DOM
```

Comment:
Technically, what happens is that changing the bound property from `false` to `false` doesn't actually change it so the `this.checked = false` assignment does literally nothing, and Angular simply does not update the binding since nothing apparently changed.

You can see it very clearly in this example: https://stackblitz.com/edit/prevent-checkbox-change?file=src%2Fapp%2Fapp.component.html
I think what you have is fine, but it has nothing to do with the link you posted so I'd rather you remove it.